### PR TITLE
chore: Improve Documentation (RMCCX-6705)

### PR DESCRIPTION
**Pull request by SoumenRautray**
# Description
Improve Documentation

## Links
[RMCCX-6705]

# Checklist
- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
- [ ] I have added to the [changelog](../blob/master/CHANGELOG.md#Unreleased)
- [ ] I wrote/updated tests for new/changed code
- [ ] I removed all sensitive data **before every commit**, including API endpoints and keys, and internal links
- [ ] I ran `fastlane ci` without errors
- [ ] All project file changes are replicated in `SampleSPM/SampleSPM.xcodeproj` project

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -97,37 +109,138 @@ func application(_ application: UIApplication, didFinishLaunchingWithOptions lau
...
+>   - The internal Backend only supports production access token
+> - Migrating from legacy Mobile Login SDK to ID SDK
+>   - Update your `UserInfoProvider` and return valid value for the `getIDTrackingIdentifier()` method. Do not provide values for other methods or leave them as null or empty.
+>   - **Impact**: User will be treated as a new user, therefore if there are **active** messages that were previously displayed/opted-out by the user, then it will be displayed again
+
+
+### **3. logEvent()**  
+This method is provided for the host application to log and save events. These events will be used to match campaign triggers which initiates the display of a message whenever a specific event or a set of events occur. Call this method at appropriate locations in your app, and based on your use-case.
+
+For each logged event, the SDK will match it with the ongoing message's triggers that are configured in the Dashboard. Once all of the required events are logged by the app, the message will be displayed in the current registered activity. If no activity is registered, it will be displayed in the next registered activity.
```
Comment (Esakkiraja-Pothikannan) [inline README.md]: what does it mean `registered activity` or `activity`?

File: README.md [hunk 2]
```diff
@@ -97,37 +109,138 @@ func application(_ application: UIApplication, didFinishLaunchingWithOptions lau
...
-This method is provided for the host application to log and save events. These events will be used to match campaign triggers.
+### <a name="register-preference"></a>2. registerPreference()
+
+A preference is what will allow IAM to identify users for targeting and segmentation. Preference object should implement `UserInfoProvider` protocol and provide any of the following identifiers (not all need to be provided):
+
+1.  UserID - An unique identifier associated with user membership. Usually it's the name used in login process (e.g. an email address).
+1.  IDTrackingIdentifier - The value provided by the internal ID SDK as the "tracking identifier" value.
+1.  AccessToken - This is the token provided by the internal RAuthentication SDK as the "accessToken" value
+
+⚠️ The method is designed to be called ONCE once per app session - i.e. only one instance of `UserInfoProvider` can be created. IAM SDK will read object's properties on demand. There's no need to call this method again after login/logout for example.
```
Comment (Esakkiraja-Pothikannan) [inline README.md]: typo: " **ONCE once** per app session"

File: README.md [hunk 3]
```diff
@@ -139,76 +252,93 @@ let attributesList = [stringAttribute, intAttribute, boolAttribute, doubleAttrib
...
-        "member-id" 
-    }
-
-    func getIDTrackingIdentifier() -> String? { nil }
-    func getAccessToken() -> String? { nil }
+func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
+    RInAppMessaging.configure(subscriptionID: "subscriptionID", 
+                                configurationURL: "configURL",
+                                enableTooltipFeature: true)
+    RInAppMessaging.registerPreference(self.preferenceData)
```
Comment (Esakkiraja-Pothikannan) [inline README.md:293]: our suggestion is to register the user before call configure method, right?

File: README.md [hunk 4]
```diff
@@ -139,76 +252,93 @@ let attributesList = [stringAttribute, intAttribute, boolAttribute, doubleAttrib
...
+    var easyID: String?
+    var accessToken: String?
 
-## **Custom Events**
+}
 
-As shown in the example above for event logging, IAM also supports custom event. Custom events are events with an unique name and a list of attributes associated with them.
+extension IAMPreferenceData {
+    func getAccessToken() -> String? {
+        "accessToken"
```
Comment (Esakkiraja-Pothikannan) [inline README.md]: I guess, we need to return the instance not String. else there is no use of declaring the properties in the `IAMPreferenceData`

File: README.md (hunk 3 above)
Comment (SoumenRautray) [reply README.md:293]: The registerPreference call will be retained incase there is delay in execution of configure(). But for best practice it should be called after configure ()

**Review by Esakkiraja-Pothikannan — APPROVED** [review]

**Pull request merged**